### ISSUE TITLE
Regarding the Compilation Time of demos/denoising.ipynb

### ISSUE THREAD

**Issue by daihuiao** [berthyf96/score_prior#3]
First of all, thank you for this amazing project!

Although the README explains that the last Jupyter notebook uses diffrax to solve for exact likelihood, when I executed the final segment of demos/denoising.ipynb, the compilation time exceeded my expectations. After about one or two hours, I noticed it still hadn't completed a single gradient update, and the logs only indicated that the compilation time was excessively long. Therefore, I would like to ask how long it typically takes to compute the exact likelihood and perform backpropagation using diffrax to complete the first step and the entire training process, and what kind of hardware configuration is required.

> 2024-06-16 16:11:22.350559: E external/xla/xla/service/slow_operation_alarm.cc:65] 
[Compiling module pmap_step_fn] Very slow compile? If you want to file a bug, run with envvar XLA_FLAGS=--xla_dump_to=/tmp/foo and attach the results.
2024-06-16 16:19:05.655567: E external/xla/xla/service/slow_operation_alarm.cc:133] The operation took 9m43.304804206s
[Compiling module pmap_step_fn] Very slow compile? If you want to file a bug, run with envvar XLA_FLAGS=--xla_dump_to=/tmp/foo and attach the results.

My CPU has been very busy all this time. Is this normal?
![image](https://github.com/berthyf96/score_prior/assets/74499474/718e1958-d45d-4925-801b-c05b571a74c0)

Thanks again for your amazing work on this project!

**Comment by daihuiao** [comment]
I can see from Issue #1 that it takes about 10 minutes to jit compile, and my model has been running for more than an hour, is it because I did not use a graphics card, because my graphics card (3080,10G) did not add video memory?
:)

**Comment by daihuiao** [comment]
I found out that the reason was because the model didn't use the graphics card properly for calculations,
```
import jax
print(jax.devices())

```

> CUDA backend failed to initialize: Found cuDNN version 8700, but JAX was built against version 8906, which is newer. The copy of cuDNN that is installed must be at least as new as the version against which JAX was built. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
